src/Planners/src/QuinticSpline.cpp

```diff
@@ -0,0 +1,851 @@
+                                            const Polynomial& poly,
+                                            Eigen::Ref<Eigen::VectorXd> velocity)
+{
+    // improve the readability of the code
+    const auto& a0 = poly.a0;
+    const auto& a1 = poly.a1;
+    const auto& a2 = poly.a2;
+    const auto& a3 = poly.a3;
+    const auto& a4 = poly.a4;
+    const auto& a5 = poly.a5;
+
+    velocity = a1
+             + 2 * a2 * t
+             + 3 * a3 * std::pow(t, 2)
+             + 4 * a4 * std::pow(t, 3)
+             + 5 * a5 * std::pow(t, 4);
+}
+
+void QuinticSpline::Impl::getAccelerationAtTime(const double& t,
+                                                const Polynomial& poly,
+                                                Eigen::Ref<Eigen::VectorXd> acceleration)
+{
+    // improve the readability of the code
+    const auto& a0 = poly.a0;
+    const auto& a1 = poly.a1;
+    const auto& a2 = poly.a2;
+    const auto& a3 = poly.a3;
+    const auto& a4 = poly.a4;
+    const auto& a5 = poly.a5;
+
+    acceleration = 2 * a2
+                 + 2 * 3 * a3 * t
+                 + 3 * 4 * a4 * std::pow(t, 2)
+                 + 4 * 5 * a5 * std::pow(t, 3);
+}
+
+bool QuinticSpline::evaluatePoint(const double& t,
+                                  Eigen::Ref<Eigen::VectorXd> position,
+                                  Eigen::Ref<Eigen::VectorXd> velocity,
+                                  Eigen::Ref<Eigen::VectorXd> acceleration)
+{
+
+    if (!m_pimpl->areCoefficientsComputed)
+    {
+        if (!m_pimpl->computeCoefficients())
+        {
+            std::cerr << "[QuinticSpline::evaluatePoint] Unable to compute the coefficients of the "
+                         "spline."
+                      << std::endl;
+            return false;
+        }
+    }
+
+    if (t < m_pimpl->polynomials.front().initialPoint->timeInstant)
+    {
+        position = m_pimpl->knots.front().position;
+        velocity = m_pimpl->knots.front().velocity;
+        acceleration = m_pimpl->knots.front().acceleration;
+
+        return true;
+    }
+
+    if (t >= m_pimpl->polynomials.back().finalPoint->timeInstant)
+    {
+        position = m_pimpl->knots.back().position;
+        velocity = m_pimpl->knots.back().velocity;
+        acceleration = m_pimpl->knots.back().acceleration;
+
+        return true;
+    }
+
+    const auto poly = std::find_if(m_pimpl->polynomials.begin(),
+                                   m_pimpl->polynomials.end(),
+                                   [&t](const auto& p) {
+                                       return ((t >= p.initialPoint->timeInstant)
+                                               && (t < p.finalPoint->timeInstant));
+                                   });
+
+    if (poly == m_pimpl->polynomials.end())
+    {
+        std::cerr << "[QuinticSpline::evaluatePoint] Unable to find the sub-trajectory."
+                  << std::endl;
+        return false;
+    }
```

Comment:
Notice that ``find_if`` performs a linear search, not a binary search. In any case, this is not a big deal with few knot points.

Reply:
We should keep in mind this if we notice performance limitations.